Source/Libraries/openXDA.DataPusher/WebAPIHub.cs

```diff
@@ -72,588 +46,321 @@ public static string CompanyName
+        public static HttpResponseMessage ProcessFileGroup(JObject record, DataPusherRequester requester)
         {
-            return ProcessFileGroup(instance, record, userAccount);
-        }
-
-        public static HttpResponseMessage ProcessFileGroup(string instance, JObject record, UserAccount userAccount)
-        {
-            Task<HttpResponseMessage> task = ProcessFileGroupAsync(instance, record, userAccount);
+            Task<HttpResponseMessage> task = ProcessFileGroupAsync(record, requester);
             return task.GetAwaiter().GetResult();
         }
 
-        public static async Task<HttpResponseMessage> ProcessFileGroupAsync(string instance, JObject record, UserAccount userAccount)
+        public static async Task<HttpResponseMessage> ProcessFileGroupAsync(JObject record, DataPusherRequester requester)
         {
-            string url = BuildURL(instance, "api/PQMark/ProcessFileGroup");
-            string antiForgeryToken = await GenerateAntiForgeryTokenAsync(instance, userAccount);
-
-            using (HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, url))
+            HttpResponseMessage response = await requester.SendRequestAsync("api/PQMark/ProcessFileGroup", HttpMethod.Post, record);
+            if (!response.IsSuccessStatusCode)
             {
-                ConfigureRequest(request, userAccount);
-                request.Headers.Add("X-GSF-Verify", antiForgeryToken);
-
-                string json = record.ToString(Formatting.None);
-                request.Content = new StringContent(json, null, "application/json");
-
-                HttpResponseMessage response = await HttpClient.SendAsync(request);
-
-                if (!response.IsSuccessStatusCode)
-                {
-                    string message = $"Server returned status code {response.StatusCode}: {response.ReasonPhrase}";
-                    response.Dispose();
-                    throw new InvalidOperationException(message);
-                }
-
-                return response;
+                string message = $"Server returned status code {response.StatusCode}: {response.ReasonPhrase}";
+                response.Dispose();
+                throw new InvalidOperationException(message);
             }
+            return response;
         }
 
         #endregion
 
-        #region [ Helpers ]
+        #region [ UserAccount Helpers ]
 
-        private static string BuildURL<T>(string baseAddress, string basePath, string whereClause) where T : class
+        private static string BuildPath<T>(string basePath, string whereClause) where T : class
         {
-            string baseURL = BuildURL<T>(baseAddress, basePath);
+            string baseURL = BuildPath<T>(basePath);
             string urlEncodedWhereClause = HttpUtility.UrlEncode(whereClause);
             return $"{baseURL}?where={urlEncodedWhereClause}";
```

Comment:
This pattern has always worried me. We should think about moving away from passing an arbitrary WHERE clause to the server as much as possible so we can retire APIs with high potential for SQL injection.

Reply:
I should also mention, this is just a general comment about `WebAPIHub` to increase awareness. There's nothing actionable here. You can resolve once you've read it.

Reply:
I too actually have been worried about this exact same thing, so we are of one mind on it. I want to fix it where possible, but it usually isn't actionable as is the case here.